external-crates/move/crates/move-binary-format/src/file_format.rs

```diff
@@ -2313,7 +2313,7 @@ impl move_abstract_interpreter::control_flow_graph::Instruction for Bytecode {
 /// A module is published as a single entry and it is retrieved as a single blob.
 #[derive(Clone, Debug, Default, Eq, PartialEq)]
 #[cfg_attr(feature = "fuzzing", derive(arbitrary::Arbitrary))]
-#[cfg_attr(feature = "wasm", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
```

Comment:
we intentionally do not add serde here for security reasons. We need to ensure that the file format is never serialized in this manner

Reply:
Maybe I miss something but could you elaborate more about the security implications here?

Reply:
It is security in the sense of not allowing mistakes to be checked into the codebase. In some ways, the types here in `file_format` are just a Rust representation of the actual source of truth, which is defined in `external-crates/move/crates/move-binary-format/src/serializer.rs` and `external-crates/move/crates/move-binary-format/src/deserializer.rs`. We do not to allow two forms of serialization/deserialization to be checked into the code base to proactively prevent any such mistake. 
As such, we cannot add this `derive` as defined here. We allowed it for `"wasm"` the more narrow scope of that feature flag.

Reply:
In that case, is it acceptable to `impl Serialize/Deserialize` with `external-crates/move/crates/move-binary-format/src/serializer.rs`? We do not care about the actual serialization format but just would like to see `Serialize/Deserialize` get implemented.